docs/proposals/20191017-control-plane.md

```diff
@@ -0,0 +1,566 @@
+    - [Graduation Criteria](#graduation-criteria)
+      - [Alpha -> Beta Graduation](#alpha---beta-graduation)
+    - [Upgrade Strategy](#upgrade-strategy)
+  - [Alternatives](#alternatives)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+The lexicon used in this document is described in more detail [here](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/reference/glossary.md).  Any discrepancies should be rectified in the main Cluster API glossary.
+
+### References
+
+[Kubernetes Control Plane Management Overview](https://docs.google.com/document/d/1nlWKEr9OP3IeZO5W2cMd3A55ZXLOXcnc6dFu0K9_bms/edit#)
+
+## Summary
+
+This proposal outlines a new process for Cluster API to manage control plane machines as a single concept. This includes
+upgrading, scaling up, and modifying the underlying image (e.g. AMI) of the control plane machines.
+
+The control plane covered by this document is defined as the Kubernetes API server, scheduler, controller manager, DNS
+and proxy services, and the underlying etcd data store.
+
+## Motivation
+
+During 2019 we saw control plane management implementations in each infrastructure provider. Much like
+bootstrapping was identified as being reimplemented in every infrastructure provider and then extracted into Cluster API
+Bootstrap Provider Kubeadm (CABPK), we believe we can reduce the redundancy of control plane management across providers
+and centralize the logic in Cluster API.
+
+### Goals
+
+- To establish new resource types for control plane management
+- To support single node and multiple control plane instances
+- To enable declarative orchestrated control plane upgrades
+- To provide a default machine-based implementation using kubeadm
+
+#### Additional goals of the default kubeadm machine-based Implementation
+
+- To provide a kubeadm-based implementation that is infrastructure provider agnostic
+- To enable declarative orchestrated replacement of control plane machines, such as to rollout an OS-level CVE fix.
+- To manage a kubeadm-based, "stacked etcd" control plane
+- To enable scaling of the number of control plane nodes
+- To support pre-existing, user-managed, external etcd clusters
+- To support only user-initiated remediation:
+  E.g. user deletes a Machine. Control Plane Provider reconciles by removing the corresponding etcd member and updating related metadata
+
+### Non-Goals / Future Work
+
+Non-Goals listed in this document are intended to scope bound the current v1alpha3 implementation and are subject to change based on user feedback over time.  
+
+- To manage non-machine based topologies, e.g.
+  - Pod based control planes; these can be managed via standard kubernetes objects.
+  - Non-node control planes (i.e. EKS, GKE, AKS); these can be managed via the respective APIs.
+- To manage CA certificates outside of what is provided by Kubeadm bootstrapping
+- To manage etcd clusters in any topology other than stacked etcd.
```

Comment:
Correct, since we aren't involved with the lifecycle of the user-managed, external etcd cluster we do not care about how it is configured or managed :)